azure-devops/azext_devops/dev/repos/import_request.py

```diff
@@ -4,33 +4,78 @@
 # --------------------------------------------------------------------------------------------
 
 import time
+from knack.prompting import prompt_pass
+from azext_devops.devops_sdk.v5_0.service_endpoint.models import ServiceEndpoint, EndpointAuthorization
 from azext_devops.devops_sdk.v5_0.git.models import GitImportRequestParameters
 from azext_devops.devops_sdk.v5_0.git.models import GitImportGitSource
 from azext_devops.devops_sdk.v5_0.git.models import GitImportRequest
 
-from azext_devops.dev.common.services import get_git_client, resolve_instance_project_and_repo
+from azext_devops.dev.common.services import (get_git_client,
+                                              resolve_instance_project_and_repo,
+                                              get_service_endpoint_client)
+from azext_devops.dev.common.const import CLI_ENV_VARIABLE_PREFIX
+from azext_devops.dev.common.prompting import verify_is_a_tty_or_raise_error
 
 
 def create_import_request(git_source_url, project=None, repository=None,
+                          requires_authorization=False,
+                          user_name=None,
+                          git_service_endpoint_id=None,
                           organization=None, detect=None):
     """Create a git import request (currently only supports import from public git source)
     :param repository: Name or ID of the repository to create the import request in.
     :type repository: str
-    :param git_source_url: Url of the source git repository
+    :param git_source_url: Url of the source git repository.
     :type git_source_url: str
+    :param requires_authorization: Flag to tell if source git repository is private.
+    :type requires_authorization: bool
+    :param user_name: User name in case source git repository is private.
+    :type user_name: str
+    :param git_service_endpoint_id: Service Endpoint for connection to external endpoint.
+    :type git_service_endpoint_id: str
     """
     organization, project, repository = resolve_instance_project_and_repo(
         detect=detect,
         organization=organization,
         project=project,
         repo=repository,
         repo_required=True)
+
+    delete_se_after_import = False
```

Comment:
Why not take this as user input?

Reply:
it's like if we create we delete it, else we don't
very much like UI (where this is not exposed as well)

Reply:
Rest API gives you an option to delete after use, so I was comparing with that. But sounds good to me to hide that param. People can always script it using the delete service endpoint command.